test/test_svg.py

```diff
@@ -1,47 +1,62 @@
 import unittest
-from lsystems.svg import PathSegment, Line, Rotation
-
-previous_segment = PathSegment(
-    start_position=(0, 0),
-    start_direction=0,
-    end_position=(10, 10),
-    end_direction=90,
-    d=""
-)
+from lsystems.svg import Line, Rotation, Arc, PopPosition, PushPosition, Cursor
 
 
 class TestSvg(unittest.TestCase):
     def test_line(self):
         line = Line(length=20)
-        segment = line.generate(previous_segment)
+        start_cursor = Cursor(x=11, y=10, dir=90)
+
+        segment, end_cursor = line.generate(start_cursor)
 
-        self.assertEqual(segment.start_position, previous_segment.end_position)
-        self.assertEqual(segment.start_direction, previous_segment.end_direction)
-        self.assertAlmostEqual(segment.end_position[0], 10)
-        self.assertAlmostEqual(segment.end_position[1], 30)
-        self.assertEqual(segment.end_direction, segment.start_direction)
-        self.assertEqual(segment.d, "L 10.0000 30.0000")
+        self.assertEqual(segment, "L 11.0000 30.0000")
+        self.assertEqual(end_cursor, Cursor(x=11, y=30, dir=90), f"Falscher Cursor: {end_cursor}")
 
     def test_line_without_draw(self):
         line = Line(length=20, draw=False)
-        segment = line.generate(previous_segment)
+        start_cursor = Cursor(x=11, y=10, dir=90)
+
+        segment, end_cursor = line.generate(start_cursor)
 
-        self.assertEqual(segment.start_position, previous_segment.end_position)
-        self.assertEqual(segment.start_direction, previous_segment.end_direction)
-        self.assertAlmostEqual(segment.end_position[0], 10)
-        self.assertAlmostEqual(segment.end_position[1], 30)
-        self.assertEqual(segment.end_direction, segment.start_direction)
-        self.assertEqual(segment.d, "M 10.0000 30.0000")
+        self.assertEqual(segment, "M 11.0000 30.0000")
+        self.assertEqual(end_cursor, Cursor(x=11, y=30, dir=90), f"Falscher Cursor: {end_cursor}")
 
     def test_rotation(self):
         line = Rotation(angle=122)
-        segment = line.generate(previous_segment)
+        start_cursor = Cursor(x=11, y=10, dir=90)
+
+        segment, end_cursor = line.generate(start_cursor)
+
+        self.assertEqual(end_cursor, Cursor(x=11, y=10, dir=90-122), f"Falscher Cursor: {end_cursor}")
```

Comment:
error-messages should maybe be in English (also below) :)

Reply:
done